### PR TITLE
Use foldl' over foldl everywhere

### PR DESCRIPTION
**Pull request by EggBaconAndSpam**
As suggested by @sjakobi. Germans banding together in the fight against
laziness!

### CODE REVIEW

**Comment by sjakobi** [comment]
Thanks for addressing this right away! :)

Do the changes in the parser and the binary decoding have a measurable performance impact?

We have parser benchmarks, and I believe we have some for binary decoding too, but I'm not quite sure.

**Comment by EggBaconAndSpam** [comment]
I can't see a significant huge difference, but I also don't trust my laptop with cpu throttling etc. Could you have a look?

File: dhall-lsp-server/src/Dhall/LSP/Backend/Completion.hs [hunk 1]
```diff
@@ -158,7 +159,7 @@ completeProjections (CompletionContext context values) expr =
   -- substitute 'dependent lets', necessary for completion of unions
   let values' = toList values
       subs = filter ((/= holeExpr) . snd) $ zip (contextToVariables values') (map snd values')
-      expr' = foldl (\e (x,val) -> subst x val e) expr subs
+      expr' = foldl' (\e (x,val) -> subst x val e) expr subs
```
Comment (sjakobi) [inline dhall-lsp-server/src/Dhall/LSP/Backend/Completion.hs]: What is `subst` here?

File: dhall/src/Dhall/Tags.hs [hunk 2]
```diff
@@ -144,7 +144,7 @@ fixPosAndDefinition t = foldMap (\(LC ln c, term) ->
           --   In example above, first line is 15 bytes long and '\n', mls contain:
           --   (1, (16, "let foo = "bar"")
           --   That allow us to get byte offset easier.
-          mls = M.fromList . fst . foldl processLine ([], 0) . zip [1..] $ T.lines t
+          mls = M.fromList . fst . foldl' processLine ([], 0) . zip [1..] $ T.lines t
```
Comment (sjakobi) [inline dhall/src/Dhall/Tags.hs]: I think this only change where I'd expect a noticeable effect.

File: dhall-lsp-server/src/Dhall/LSP/Backend/Completion.hs (hunk 1 above)
Comment (EggBaconAndSpam) [reply dhall-lsp-server/src/Dhall/LSP/Backend/Completion.hs]: Plain substitution: https://hackage.haskell.org/package/dhall-1.27.0/docs/Dhall-Core.html#v:subst

**Comment by EggBaconAndSpam** [comment]
What are everybody else's opinions on this? Yay? Nay? Don't give a damn?

**Pull request merged**